### DISCUSSION TITLE
Why receipt.dat file have the same content of the peppol document?

### DISCUSSION THREAD

**Discussion by francescodiperna** [OxalisCommunity/Oxalis-AS4#221] [General]
Hi, when a peppol document is received we got two files:

- XXXXXXXXXXXXXXXXXXXXXXXX.doc.xml
- XXXXXXXXXXXXXXXXXXXXXXXX.receipt.dat

With the same content inside.
Is it normal?

Below my configuration:
```

oxalis.keystore: {
    path: oxalis-keystore.jks
    password: "ZZZZZZZZZZZZZZZZZZz"

    key: {
        alias: "pit000479 (peppol access point ca - g2)"
        password: "ZZZZZZZZZZZZZZZZZZz"
    }
}

# The folder where the incoming messages will be stored
oxalis.path.inbound = /azshare/inbox/
oxalis.path.outbound = /azshare/outbox/

# Optional
oxalis.logging.config=/oxalis/conf/logback.xml


oxalis.pki.version=V2

# The relative name of the directory holding plugin
oxalis.path.plugin = plugin


#oxalis.path.conf = ""

# Signals to Oxalis that we should look for plugin
oxalis.persister.receipt = default
oxalis.persister.payload = default

```

Best regards

**Comment by dladlk** [comment]
Hello, it is not normal. Do you mean that your *.receipt.dat file contains payload instead of DeliveryNonDeliveryToRecipient XML?
According to https://github.com/OxalisCommunity/oxalis/blob/ebaf74463a963860be5a7475eeb342f3d0ebd7e5/oxalis-commons/src/main/java/network/oxalis/commons/persist/DefaultPersister.java#L80 - *.receipt.dat should contain result of evidency factory invokation.

You can try to activate debug logging on DefaultPersister to see if you get log.debug("Receipt persisted to: {}", path); - and also check which implementation of evidence factory is configured...

**Reply by francescodiperna** [reply]
Thanks for reply, exactly, the receip.dat file contain the payload.

**Reply by francescodiperna** [reply]
Hi, 
I'm using this image "norstella/oxalis-as4:6.0.0". I do not see the debug logging you are talking before. And also the evendence factory (how to check this?).

bye
Francesco

**Reply by dladlk** [reply]
You should have logback.xml in your classpath with the line like 

`<logger name="network.oxalis.commons.persist" level="debug"/>
`

It is very strange to have the same file in receipt as in payload - I would triple check that it is so and that no other app then Oxalis overwrites it. Receipt persistence is done via another interface, I am curious how it can be misconfigured...

**Reply by francescodiperna** [reply]
Thanks very much Dmytro Lapko, with your help I found the bug.
Now I can see the receipt.

Bye

**Reply by dladlk** [reply]
Great! Can you describe what was wrong? It is most interesting :)

**Reply by francescodiperna** [reply]
Good morning, it wasn't an error of oxalis but the software that elaborate the files received.
Bye
